google/cloud/bigquery/dataset.py

```diff
@@ -898,6 +900,29 @@ def storage_billing_model(self, value):
             )
         self._properties["storageBillingModel"] = value
 
+    @property
+    def external_catalog_dataset_options(self):
+        """Options defining open source compatible datasets living in the
+        BigQuery catalog. Contains metadata of open source database, schema
+        or namespace represented by the current dataset."""
+
+        prop = _helpers._get_sub_prop(
+            self._properties, ["externalCatalogDatasetOptions"]
+        )
+
+        if prop is not None:
+            prop = external_config.ExternalCatalogDatasetOptions().from_api_repr(prop)
```

Comment:
`from_api_repr` should be a classmethod, so no instance is needed.

```suggestion
            prop = external_config.ExternalCatalogDatasetOptions.from_api_repr(prop)
```